daemon/SecretsImpl/secretsrequestprocessor.cpp

```diff
@@ -6394,3 +6394,46 @@ void Daemon::ApiImpl::RequestProcessor::authenticationCompleted(
     }
 }
 
+void Daemon::ApiImpl::RequestProcessor::cancelRequest(pid_t callerPid,
+                                                      quint64 requestId)
+{
+    // look up the pending request in our list
+    if (m_pendingRequests.contains(requestId)) {
+        // call the appropriate method to complete the request
+        Daemon::ApiImpl::RequestProcessor::PendingRequest pr = m_pendingRequests.take(requestId);
+        Q_ASSERT(pr.callerPid == callerPid);
+        switch (pr.requestType) {
+        case UserInputRequest: {
+            if (pr.parameters.size() != 1) {
+                qCWarning(lcSailfishSecretsDaemon) << QLatin1String("Internal error: incorrect parameter count!");
+            } else {
+                QString userInputPlugin = pr.parameters.takeFirst().value<QString>();
+                if (!m_authenticationPlugins.contains(userInputPlugin)) {
+                    qCWarning(lcSailfishSecretsDaemon) << QLatin1String("Internal error: unknown plugin name!");
+                } else {
+                    m_authenticationPlugins[userInputPlugin]->cancelUserInputInteraction(callerPid, requestId);
+                }
+            }
+            break;
+        }
+        case DeleteCollectionRequest:
+        case StoredKeyIdentifiersRequest:
+        case SetCollectionSecretRequest:
+        case GetCollectionSecretRequest:
+        case GetStandaloneSecretRequest:
+        case FindCollectionSecretsRequest:
+        case DeleteCollectionSecretRequest:
+        case UseCollectionKeyPreCheckRequest:
+        case SetCollectionKeyPreCheckRequest: {
+            // always use the system authentication plugin for device lock authentication requests.
```

Comment:
But if the collection is custom locked, they may not have used the system authentication plugin, unless I'm misunderstanding?

Reply:
Well, grepping beginAuthentication calls in the daemon (which are the ones we want to cancel in this switch entries), return only calls from system authentication plug-in. See result of `grep -r beginAuthentication daemon/`. That's why, I copy-pasted the code from all these places to get the same plugin.

If in the future, we allow custom authentication, we will need to store the plugin name, as done for the user input request above.